### DISCUSSION TITLE
Not sure how to use the wieght attribute in the NeighborLoader

### DISCUSSION THREAD

**Discussion by atul1234anand** [pyg-team/pytorch_geometric#9177] [Q&A] [answered]
Hi. I tried using the NeighborLoader as mentioned in [this post](https://github.com/pyg-team/pytorch_geometric/discussions/8290). However, I keep getting this error that `TypeError:` __init__() got an unexpected keyword argument 'weight_attr'` . Not sure where I am going wrong. 

The following is the code I wrote

```
  loader = NeighborLoader(
      graph,
      # Sample 30 neighbors for each node for 2 iterations
      num_neighbors=[5] * 10,
      # Use a batch size of 128 for sampling training nodes
      batch_size=128,
      weight_attr='edge_weight',
      shuffle=True
  )
```

I am using Pytorch Geometric 2.2.0 with Pytorch 1.13.1 and CUDA 11.8.0.

Request some help

**Comment by rusty1s** [comment, marked as answer]
`weight_attr` support was added in PyG 2.4 onwards.

**Reply by atul1234anand** [reply]
How do I install PyG 2.4? The website has instructions only for 2.2.

**Reply by rusty1s** [reply]
Do you mean PyTorch 2.2? You can simply install PyG either via
```
pip install --upgrade torch-geometric
```
or by fixing the version
```
pip install torch-geometric==2.4.0
